### PR TITLE
Restrictions: answer controller

### PR DESCRIPTION
**Pull request by Anlord241**
image

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/exceptions/ControllerAdvice.java [hunk 1]
```diff
@@ -58,4 +58,10 @@ public TestsLimitExceededResponse handleTestsLimitExceededException(TestsLimitEx
         }
         return handleExceptionInternal(e, body, new HttpHeaders(), HttpStatus.BAD_REQUEST, request);
     }
+
+    @ExceptionHandler(value = {DoNotHaveRightsException.class})
+    @ResponseStatus(value = HttpStatus.FORBIDDEN)
+    public ErrorResponse handleDoNotHaveRightsException(DoNotHaveRightsException e) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/exceptions/ControllerAdvice.java]: Why not to use [AccessControlException](https://docs.oracle.com/javase/7/docs/api/java/security/AccessControlException.html)?

File: src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java [hunk 2]
```diff
@@ -49,10 +49,22 @@ public String getUrl(Long testId, Long questionId) {
...
-        Question question = questionService.getQuestionByTestIdAndModule(testId, module);
-        return save(testId, question.getId(), url);
+
+        Question question = questionService.getQuestionByTestIdAndModule(testId, Modules.SPEAKING);
+
+        FileAnswer fileAnswer = FileAnswer.builder()
+            .id(new TestQuestionID(test, question))
+            .url(url)
+            .build();
+        return fileAnswerRepository.save(fileAnswer);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java]: Will it also work properly with update?

File: src/main/java/com/team4/testingsystem/services/impl/TestsServiceImpl.java [hunk 3]
```diff
@@ -253,4 +254,19 @@ public void deassignCoach(long id) {
             throw new TestNotFoundException();
         }
     }
+
+    @Override
+    public void checkRights(Test test) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/impl/TestsServiceImpl.java]: checkRights name is too wide while it checks only test owner. Let's rename to smth more concreate

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java (hunk 2 above)
Comment (Anlord241) [reply src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java]: Why not?
We rewrite file answer if it already exists

Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/services/impl/FileAnswerServiceImpl.java]: We expected the behavior, but seams like something is broken there

**Review by AntonMaksimenkau — APPROVED** [review]

File: src/main/java/com/team4/testingsystem/controllers/AnswerController.java [hunk 4]
```diff
@@ -32,6 +32,7 @@ public String downloadEssay(@PathVariable Long testId) {
 
     @ApiOperation(value = "Upload essay text")
     @PostMapping("/essay/{testId}")
+    @Secured("ROLE_COACH")
```
Comment (maxim12321) [inline src/main/java/com/team4/testingsystem/controllers/AnswerController.java]: Maybe this restriction is for `downloadEssay`?

**Review by maxim12321 — CHANGES_REQUESTED** [review]

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**